File: PYTHON/DAY2/TestTaskone.py
import unittest

class TestTaskOne(unittest.TestCase):
    def test_calculate_years_in_the_past(self):
        father_age = 40
        son_age = 30
        expected = 20
        self.assertEqual(expected, TestTaskone.calculate_years(fatherAge, sonAge))

    def test_calculate_years_in_the_future(self):
        father_age = 45
        son_age = 15
        expected = 15
        self.assertEqual(expected, TestTaskone.calculate_years(fatherAge, sonAge))

    def test_calculate_years_currently(self):
        father_age = 40
        son_age = 20
        expected = 0
        self.assertEqual(expected, TestTaskone.calculate_years(fatherAge, sonAge))

    def test_calculate_years_equal_ages(self):
        father_age = 0
        son_age = 0
        expected = 0
        self.assertEqual(expected, TestTaskone.calculate_years(fatherAge, sonAge))
    
    def test_calculate_years_for_zero_ages(self):
        father_age = 0
        son_age = 0
        expected = 0
        self.assertEqual(expected, TestTaskone.calculate_years(fatherAge, sonAge))

File: JAVA/DAY2/TestTaskone.java
import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.assertEquals;

public class TestTaskone {

@Test
public void testCalculateYearsInThePast() {
int fatherAge = 40;
int sonAge = 30;
int expected = 20;
        assertEquals(expected, Taskone.calculateYears(fatherAge, sonAge));
}

@Test
public void testCalculateYearsInTheFuture(){
int fatherAge = 45;
int sonAge = 15;
int expected = 15;
	assertEquals(expected, Taskone.calculateYears(fatherAge, sonAge));
}

@Test
public void testCalculateYearsCurrently() {
int fatherAge = 40;
int sonAge = 20;
int expected = 0;
	assertEquals(expected, Taskone.calculateYears(fatherAge, sonAge));
}

@Test
public void testCalculateYearsEqualAges() {
int fatherAge = 40;
int sonAge = 40;
int expected = 40;
	assertEquals(expected, Taskone.calculateYears(fatherAge, sonAge));
}

@Test
public void testCalculateYearsForZeroAges() {
int fatherAge = 0;
int sonAge = 0;
int expected = 0;
	assertEquals(expected, Taskone.calculateYears(fatherAge, sonAge));

}
}


